application/modules/admin/controllers/sessions.php

```diff
@@ -63,33 +63,33 @@ private function getNickname($data)
 
 	private function getBrowser($u_agent)
 	{
-		if(preg_match('/MSIE/i',$u_agent) && !preg_match('/Opera/i',$u_agent)) 
-		{ 
+		if(preg_match('/trident/i',$u_agent) && !preg_match('/opera/i',$u_agent))
+		{
 			return "ie";
-		} 
-		elseif(preg_match('/firefox/i',$u_agent)) 
-		{ 
+		}
+		elseif(preg_match('/opera/i',$u_agent))
```

Comment:
Did you have to move Opera?

Reply:
yes those were some special notes I read about them somewhere :
        // Special Notes:
        // (1) Opera must be checked before FireFox due to the odd
        //     user agents used in some older versions of Opera
        // (2) WebTV is strapped onto Internet Explorer so we must
        //     check for WebTV before IE
        // (3) Because of Internet Explorer 11 using
        //     "Mozilla/5.0 ([...] Trident/7.0; rv:11.0) like Gecko"
        //     as user agent, tests for IE must be run before any
        //     tests checking for "Mozilla"
        // (4) (deprecated) Galeon is based on Firefox and needs to be
        //     tested before Firefox is tested
        // (5) OmniWeb is based on Safari so OmniWeb check must occur
        //     before Safari
        // (6) Netscape 9+ is based on Firefox so Netscape checks
        //     before FireFox are necessary
        // (7) Microsoft Edge must be checked before Chrome and Safari